indico/modules/events/editing/client/js/components/EditableTypeList.jsx

```diff
@@ -0,0 +1,126 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import editableTypesURL from 'indico-url:event_editing.api_editable_types';
+
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Checkbox, Form, Loader} from 'semantic-ui-react';
+import {Form as FinalForm} from 'react-final-form';
+import {Param, Translate} from 'indico/react/i18n';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {FinalCheckbox, FinalSubmitButton, handleSubmitError} from 'indico/react/forms';
+import {indicoAxios} from 'indico/utils/axios';
+
+import './EditableTypeList.module.scss';
+
+export default function EditableTypeList({eventId}) {
+  const [editMode, setEditMode] = useState(false);
+  const {data: editableTypes, loading: isLoadingEditableTypes, reFetch} = useIndicoAxios({
+    url: editableTypesURL({confId: eventId}),
+    camelize: true,
+    trigger: eventId,
+  });
+
+  if (isLoadingEditableTypes) {
+    return <Loader inline="centered" active />;
+  } else if (!editableTypes) {
+    return null;
+  }
+
+  const handleSubmit = async formData => {
+    const url = editableTypesURL({confId: eventId});
+    const enabledTypes = Object.keys(formData).filter(name => formData[name]);
+    try {
+      await indicoAxios.post(url, {editable_types: enabledTypes});
+      reFetch();
+    } catch (e) {
+      return handleSubmitError(e);
+    }
+    setEditMode(false);
+  };
+
+  return (
+    <div styleName="editable-list">
+      <h3>List of editable types</h3>
+      <FinalForm
+        onSubmit={handleSubmit}
+        subscription={{submitting: true}}
+        initialValues={Object.assign(...editableTypes.map(([id, {enabled}]) => ({[id]: enabled})))}
+      >
+        {fprops => (
+          <>
+            <Form id="editable-type-form" onSubmit={fprops.handleSubmit}>
+              <div className="action-box">
+                {!editMode &&
+                  editableTypes.map(([id, {title, enabled}]) => {
+                    return enabled ? (
+                      <div key={id} className="section">
+                        <span className="icon icon-file" />
+                        <div className="text">
+                          <div className="label">
+                            <Param name="title" value={title} />
+                          </div>
+                        </div>
+                        <div className="toolbar">
+                          <a className="i-button icon-list">
```

Comment:
Shouldn't we use SUI buttons here since it'll look inconsistent with the buttons in edit view otherwise?

Reply:
Since we are already using action-boxes in this page I think it's not too bad to use the old buttons since they integrate well with all the old css used by the boxes. Unless we are planning to re-do the entire settings page in semantic-ui style, which I don't think is the case

Reply:
ok, makes sense. i don't think we want to spend on redoing that now...